gce/firewall.tf

```diff
@@ -12,46 +12,39 @@ resource "google_compute_firewall" "ssh" {
   }
 }
 
-resource "google_compute_firewall" "boshbosh" {
-  name = "${var.env}-boshbosh"
-  description = "SSH from trusted external sources"
+
+
+# TODO: restrict this better, opening to ports 4222, 6868, 25250, 25555, 25777 is not enough
+resource "google_compute_firewall" "internal" {
+  name = "${var.env}-cf-internal"
+  description = "Open internal communication between instances"
   network = "${google_compute_network.bastion.name}"
 
-  source_tags = [ "bosh" ]
+  source_ranges = [ "${var.bastion_cidr}", "${google_compute_address.bosh.address}/32",
+                    "${google_compute_instance.bastion.network_interface.0.access_config.0.nat_ip}/32",
+                    "${google_compute_instance.bastion.network_interface.0.address}/32" ]
   target_tags = [ "bosh" ]
 
   allow {
     protocol = "tcp"
   }
-}
-
-resource "google_compute_firewall" "boshdns" {
-  name = "${var.env}-cf-boshdns"
-  description = "Allow anything/anywhere to hit port 53 on machines tagges as bosh"
-  network = "${google_compute_network.bastion.name}"
-  source_ranges = [ "0.0.0.0/0" ]
-  target_tags = [ "bosh" ]
   allow {
     protocol = "udp"
     ports = [ 53 ]
   }
-
 }
 
-resource "google_compute_firewall" "bosh-nat" {
-  name = "${var.env}-cf-microbosh-nat"
-  description = "SSH and Bosh ports from trusted external sources"
+# TODO: check if you can restrict this better
+resource "google_compute_firewall" "haproxy" {
+  name = "${var.env}-cf-haproxy"
+  description = "Make haproxy server reachable externally"
   network = "${google_compute_network.bastion.name}"
+
   source_ranges = [ "0.0.0.0/0" ]
```

Comment:
The haproxy is open to all the world, not restricted to the office. In Tsuru we are currently limiting this. This means that anybody can connect to the deployed apps but also the CF api, loggregator, uua, services.

Speaking with @mtekel, the reason to open this is that the internal services from CF try to connect to the uua services using the CF main domain (so, via haproxy routes). As we don't know the public IP  of the VMs of these services, we cannot whitelist it. 

We can workaround this by overwriting the `uua.url` property in the specific jobs to point to an internal address. For instance, in the `loggregator_traffic_controller` which [uses this property here](https://github.com/cloudfoundry/loggregator/blob/develop/bosh/jobs/loggregator_trafficcontroller/templates/loggregator_trafficcontroller.json.erb#L25)